testutil/testclient/testeventsquery/connection.go

```diff
@@ -47,3 +48,66 @@ func NewOneTimeMockDialer(
 
 	return dialerMock
 }
+
+// NewTwiceMockConnAndDialer returns a new mock connection and mock dialer that
+// will return the mock connection when DialContext is called. The mock dialer
+// will expect DialContext to be called any times. The connection mock will
+// expect Close and Send to be called exactly twice.
+func NewTwiceMockConnAndDialer(
+	t *testing.T,
+	connClosed *atomic.Bool,
+	delayEvent *atomic.Bool,
+) (*mockclient.MockConnection, *mockclient.MockDialer) {
+	connMock := NewTwiceCloseSendConnectionMock(t, connClosed, delayEvent)
+	dialerMock := NewAnyTimesMockDailer(t, connMock)
+	return connMock, dialerMock
+}
+
+// NewTwiceCloseSendConnectionMock returns a mock connection that will expect
```

Comment:
What's so special about the number 2? I'm trying to understand why we don't parameterize it.

Reply:
I can make it a parameter 👍🏼

Reply:
@Olshansk in this particular case, I would agree so long as the function were renamed to something like `NewNTimesReconnectConnectionMock`. @h5law wdyt?

Reply:
@Olshansk in general though, I still want to resist this tendency to avoid arbitrarily coupling expectations and/or exploding the signature sizes of these testutils.